src/mongocrypt-kms-ctx.c

```diff
@@ -1548,10 +1635,11 @@ _mongocrypt_kms_ctx_init_kmip_get (mongocrypt_kms_ctx_t *kms_ctx,
    return ret;
 }
 
-const char *
+static const char *
 set_and_ret (const char *what, uint32_t *len)
 {
    if (len) {
+      BSON_ASSERT_PARAM (what);
```

Comment:
```suggestion
   BSON_ASSERT_PARAM (what);

   if (len) {
```
Assert `what` is not-null unconditionally.